transports/httpclient/src/main/java/org/commonjava/maven/galley/transport/htcli/internal/HttpDownload.java

```diff
@@ -137,7 +142,25 @@ private void writeTarget()
             try
             {
                 final HttpEntity entity = response.getEntity();
-
+                /*
+                * If the server responded with 200 but the content length doesn't match,
+                * we will delete the file and return a 502 Gateway Error status to indicate
+                * the upstream server is having problems.
+                */
+                Header[] headers = response.getHeaders( "Content-Length" );
+                Collection<Integer> successStatuses = Collections.singleton( HttpStatus.SC_OK );
+                if ( headers[0] !=null && !headers[0].getValue().equals( "" ))
+                {
+                    long contentLength = Long.parseLong( headers[0].getValue().toString() );
+                    long targetLength = target.length();
+                    if ( contentLength != targetLength && successStatuses.contains( response.getStatusLine().getStatusCode() ) )
```

Comment:
What if the content length header isn't provided? I think we need a null check here too.